packages/itwin/tree-widget/src/test/trees/stateless/models-tree/internal/ModelsTreeVisibilityHandler.test.ts

```diff
@@ -0,0 +1,2266 @@
+                  expect(status.state).to.eq(props.visibility, `Element ${elementId} has unexpected visibility`);
+                }),
+              );
+            }),
+          );
+        }),
+      );
+      return toVoidPromise(obs);
+    }
+
+    describe("subject", () => {
+      let node: HierarchyNode;
+
+      before(() => (node = createSubjectHierarchyNode(iModelConnection.elements.rootSubjectId)));
+
+      it("by default all subject models, categories and elements are hidden", async () => {
+        await Promise.all([
+          expect(handler.getVisibilityStatus(node)).to.eventually.include({ state: "hidden" }),
+          assertModelVisibility({ viewportVisibility: false, handlerVisibility: "hidden" }),
+          assertCategoryVisibility({ handlerVisibility: "hidden" }),
+          assertElementsVisibility({ visibility: "hidden" }),
+        ]);
+      });
+
+      it("showing it makes it, all its models, categories and elements visible", async () => {
+        await handler.changeVisibility(node, true);
+        await Promise.all([
+          expect(handler.getVisibilityStatus(node)).to.eventually.include({ state: "visible" }),
+          assertModelVisibility({ viewportVisibility: true, handlerVisibility: "visible" }),
+          assertCategoryVisibility({ perModelVisibilityOverride: true, handlerVisibility: "visible" }),
+          assertElementsVisibility({ visibility: "visible" }),
+        ]);
+      });
+    });
+
+    describe("model", () => {
+      it("by default it, all its categories and elements are hidden", async () => {
+        await Promise.all([
+          assertModelVisibility({ viewportVisibility: false, handlerVisibility: "hidden", modelId: firstModelId }),
+          assertCategoryVisibility({ handlerVisibility: "hidden", modelIdFilter: firstModelId }),
+          assertElementsVisibility({ visibility: "hidden", modelIdFilter: firstModelId }),
+        ]);
+      });
+
+      it("showing it makes it, all its categories and elements visible", async () => {
+        await handler.changeVisibility(createModelHierarchyNode(firstModelId), true);
+        await Promise.all([
+          assertModelVisibility({ viewportVisibility: true, handlerVisibility: "visible", modelId: firstModelId }),
+          assertCategoryVisibility({ perModelVisibilityOverride: true, handlerVisibility: "visible", modelIdFilter: firstModelId }),
+          assertElementsVisibility({ visibility: "visible", modelIdFilter: firstModelId }),
+        ]);
+      });
+
+      it("gets partial when it's visible and elements are added to never drawn list", async () => {
+        await handler.changeVisibility(createModelHierarchyNode(firstModelId), true);
+
+        const categoryId = getFirstValue(models.get(firstModelId)!);
+        const element = getFirstValue(categoryParentElements.get(categoryId)!);
+        viewport.setNeverDrawn(new Set([element]));
+        viewport.renderFrame();
+        await new Promise((r) => setTimeout(r, 30));
```

Comment:
use `waitFor` instead

Reply:
This timeout wasn't necessary at all. Removed